storages/idb-storage/src/lib.rs

```diff
@@ -0,0 +1,454 @@
+        let schemas = store.get_all(None, None).await.err_into()?;
+
+        schemas
+            .into_iter()
+            .map(|v| serde_wasm_bindgen::from_value(v).err_into())
+            .collect::<Result<Vec<Schema>>>()
+    }
+
+    async fn fetch_schema(&self, table_name: &str) -> Result<Option<Schema>> {
+        let transaction = self
+            .database
+            .transaction(&[SCHEMA_STORE], TransactionMode::ReadOnly)
+            .err_into()?;
+
+        let store = transaction.object_store(SCHEMA_STORE).err_into()?;
+        let schema = store.get(JsValue::from_str(table_name)).await.err_into()?;
+        let schema = schema
+            .map(|v| serde_wasm_bindgen::from_value(v).err_into())
+            .transpose()?;
+
+        transaction.commit().await.err_into()?;
+        Ok(schema)
+    }
+
+    async fn fetch_data(&self, table_name: &str, target: &Key) -> Result<Option<DataRow>> {
+        let column_defs = self
+            .fetch_schema(table_name)
+            .await?
+            .and_then(|schema| schema.column_defs);
+        let transaction = self
+            .database
+            .transaction(&[table_name], TransactionMode::ReadOnly)
+            .err_into()?;
+
+        let store = transaction.object_store(table_name).err_into()?;
+
+        let key: Value = target.clone().into();
+        let key: JsonValue = key.try_into()?;
+        let key = JsValue::from_serde(&key).err_into()?;
+        let row = store.get(key).await.err_into()?;
+
+        transaction.commit().await.err_into()?;
+
+        row.map(|row| convert(row, column_defs.as_deref()))
+            .transpose()
+    }
+
+    async fn scan_data(&self, table_name: &str) -> Result<RowIter> {
+        let column_defs = self
+            .fetch_schema(table_name)
+            .await?
+            .and_then(|schema| schema.column_defs);
+        let transaction = self
+            .database
+            .transaction(&[table_name], TransactionMode::ReadOnly)
+            .err_into()?;
+
+        let store = transaction.object_store(table_name).err_into()?;
+        let cursor = store
+            .open_cursor(None, Some(CursorDirection::Next))
+            .await
+            .err_into()?;
+
+        let mut cursor = match cursor {
+            Some(cursor) => cursor,
+            None => {
+                return Ok(Box::new(Vec::new().into_iter()));
+            }
+        };
+
+        let mut rows = Vec::new();
+        let mut current_key = cursor.key().err_into()?;
+        let mut current_row = cursor.value().err_into()?;
+
+        while !current_key.is_null() {
+            let key: JsonValue = current_key.into_serde().err_into()?;
+            let key: Key = Value::try_from(key)?.try_into()?;
+
+            let row = convert(current_row, column_defs.as_deref())?;
+
+            rows.push((key, row));
+
+            let result = cursor.advance(1).await;
+            if result.is_err() {
+                break;
+            }
```

Comment:
```suggestion
            if let Err(err) = result {
                match err {
                    idb::Error::CursorFinished => break,
                    _ => // temp: return Err(err.into()),
                }
            }
```

When a variant other than `CursorFinished` comes out, I think we need to propagate an error.

Reply:
Alas, I missed it in the double check, but I need to do error propagation here.